lib/semantics/semantics.cc

```diff
@@ -203,34 +203,41 @@ void SemanticsContext::PopConstruct() {
   constructStack_.pop_back();
 }
 
-void SemanticsContext::SayDoVarRedefine(
-    const parser::CharBlock &location, const Symbol &variable) {
-  const parser::CharBlock doLoc{GetDoVariableLocation(variable)};
-  CHECK(doLoc != parser::CharBlock{});
-  Say(location, "Cannot redefine DO variable '%s'"_err_en_US, variable.name())
-      .Attach(doLoc, "Enclosing DO construct"_en_US);
-}
-
-void SemanticsContext::CheckDoVarRedefine(
-    const Symbol &variable, const parser::CharBlock &location) {
+void SemanticsContext::CheckDoVarRedefine(const parser::CharBlock &location,
+    const Symbol &variable, const parser::MessageFixedText message) {
   if (const Symbol * root{GetAssociationRoot(variable)}) {
     if (IsActiveDoVariable(*root)) {
-      SayDoVarRedefine(location, *root);
+      const parser::CharBlock doLoc{GetDoVariableLocation(*root)};
```

Comment:
We don't normally declare local variables `const`.

Reply:
Thanks, Tim.  I'll change my recent declarations of local variables.